mcs/class/corlib/Test/System.Reflection/FieldInfoTest.cs

```diff
@@ -1371,6 +1371,7 @@ public void SetValueDirect ()
 			Assert.AreEqual ("B", fields.str);
 		}
 
+#if !DISABLE_REMOTING
```

Comment:
In general, is there reason to exclude these tests with #if as opposed to using `Category ("Remoting")` and excluding that by filter passed to NUnit?

Reply:
My general policy is that passing incorrect include / exclude parameters to NUnit should be "safe" in the way that you'd get failing tests, but not crash the test runner with an assertion in native code.

I tried to use `[Category]` attributes as much as possible in this PR, but there are two situations here where `#if` conditionals were required:

1.) Code that wouldn't even compile due to remoting APIs being removed.
2.) Code that when run (by accident, ie. by passing incorrect flags to NUnit) would cause a native assertion.

The second situation happens when you build the Mono runtime with remoting support and then use a corlib that has certain types removed.

In theory, there shouldn't be any API entry points where this might happen, but I've seen it in a few places.